registry/quilt_server/auth.py

```diff
@@ -123,7 +123,6 @@ def check_conflicts(username, email):
 
     try:
         db.session.add(user)
-        db.session.commit()
     except IntegrityError:
```

Comment:
Pretty sure this won't be triggered just by `.add(...)` - that one is entirely client-side. If you remove the commit, then need at least a `db.session.flush()`.

That said, there's still a race condition, and also, I really think `_create_user` should only be creating new users, and not modifying existing ones.